lb_content_resolver/formats/flac.py

```diff
@@ -15,24 +16,24 @@ def get(tags, tag, default):
     return t[0]
 
 
-def read(file, mtime):
-    mdata = { 'mtime' : mtime }
+def read(file, mtime, unknown_string):
+    mdata = { "mtime" : mtime }
 
     tags = None
     try:
         tags = mutagen.flac.FLAC(file)
     except mutagen.flac.HeaderNotFoundError:
-        print("Cannot read metadata from file %s" % file.encode('utf-8'))
+        print("Cannot read metadata from file %s" % file.encode("utf-8"))
         return None
 
-    mdata['duration'] = int(tags.info.length * 1000)
-    mdata['artist_name'] = get(tags, 'artist', unknown_str)
-    mdata['artist_sortname'] = get(tags, 'artistsort', mdata['artist_name'])
-    mdata['release_name'] = get(tags, 'album', unknown_str)
-    mdata['recording_name'] = get(tags, 'title', unknown_str)
-    mdata['track_num'] = int(get(tags, 'tracknumber', "0"))
-    mdata['artist_mbid'] = get(tags, 'musicbrainz_artistid', "")
-    mdata['recording_mbid'] = get(tags, 'musicbrainz_releasetrackid', "")
-    mdata['release_mbid'] = get(tags, 'musicbrainz_albumartistid', "")
+    mdata["duration"] = int(tags.info.length * 1000)
+    mdata["artist_name"] = get(tags, "artist", unknown_string)
+    mdata["artist_sortname"] = get(tags, "artistsort", mdata["artist_name"])
+    mdata["release_name"] = get(tags, "album", unknown_string)
+    mdata["recording_name"] = get(tags, "title", unknown_string)
+    mdata["track_num"]  = TagUtils.extract_track_number(get(tags, "tracknumber", "0"))
```

Comment:
This logic was needed in both formats and probably in others so I moved it out into a util class.